gocourse03/main.go

```diff
@@ -0,0 +1,143 @@
+package main
+
+import (
+	"fmt"
+	"time"
+)
+
+const (
+	Chip        RodentType = "Chip"
+	Dale        RodentType = "Dale"
+	Cheaser     RodentType = "Cheaser"
+	Hackwrench  RodentType = "Hackwrench"
+	NorthSector Sector     = "North"
+	SouthSector Sector     = "South"
+	WestSector  Sector     = "West"
+	EastSector  Sector     = "East"
+	Center      Sector     = "Center"
+)
+
+type RodentType string
+
+type Sector string
+
+func main() {
+	var rodents [4]Rodent
+	var westRodents []Rodent
+	var eastRodents []Rodent
+	var northRodents []Rodent
+	var southRodents []Rodent
+	var movements []Movement
+
+	rodents = [4]Rodent{
+		{ID: 1, Sector: &westRodents, RodentType: Chip, FromTo: [2]Sector{WestSector, WestSector}},
+		{ID: 2, Sector: &eastRodents, RodentType: Dale, FromTo: [2]Sector{EastSector, EastSector}},
+		{ID: 3, Sector: &southRodents, RodentType: Cheaser, FromTo: [2]Sector{SouthSector, SouthSector}},
+		{ID: 4, Sector: &northRodents, RodentType: Hackwrench, FromTo: [2]Sector{NorthSector, NorthSector}},
+	}
+
+	westRodents = append(westRodents, rodents[0])
+	eastRodents = append(eastRodents, rodents[1])
+	southRodents = append(southRodents, rodents[2])
+	northRodents = append(northRodents, rodents[3])
+
+	PrintLocations(rodents, movements)
+
+	moveThroughtCenter(1, time.Now().Add(10), WestSector, EastSector, movements, rodents, westRodents, eastRodents, northRodents, southRodents)
+	moveThroughtCenter(2, time.Now().Add(11), EastSector, NorthSector, movements, rodents, westRodents, eastRodents, northRodents, southRodents)
+	moveThroughtCenter(2, time.Now().Add(12), NorthSector, SouthSector, movements, rodents, westRodents, eastRodents, northRodents, southRodents)
+	moveThroughtCenter(3, time.Now().Add(13), SouthSector, WestSector, movements, rodents, westRodents, eastRodents, northRodents, southRodents)
+
+	PrintLocations(rodents, movements)
+}
+
+type Rodent struct {
+	ID         int
+	Sector     *[]Rodent
```

Comment:
Не можу зрозуміти, чому `Sector` має тип поінтер на слайс. Зазвичай, можна обійтись просто слайсом.

Чи можемо переписати ось так, чи є якась прихована задумка?
```suggestion
	Sector     []Rodent
```

Reply:
якшо без поінтера то це будуть просто різні слайси, я так і зробив спочатку, яале потім зрозумів шо в мене дані не змінюються

Reply:
вирішив взагалі видалити цю ссилку, вона тут посуті не потрібна